File: update_menus.py
import os
import re

# 最新のメニューブロック
NEW_MENU = """    <div class="hamburger-menu-content" id="hamburgerMenu">
      <a href="index.html">トップページ</a>
      <a href="schedule.html">全国のライブスケジュール一覧</a>
      <a href="live.html">現在演奏中のミュージシャン</a>
      <a href="artists.html">アーティスト一覧</a>
      <a href="admin_listener.html">マイページ</a>
    </div>"""

def update_html_files(directory):
    for root, dirs, files in os.walk(directory):
        # node_modulesなどは除外
        if 'node_modules' in dirs:
            dirs.remove('node_modules')
        if '.git' in dirs:
            dirs.remove('.git')
            
        for file in files:
            if file.endswith('.html'):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                # hamburger-menu-contentのブロックを置換
                # <div class="hamburger-menu-content" id="hamburgerMenu"> ... </div> を探す
                pattern = re.compile(r'<div class="hamburger-menu-content" id="hamburgerMenu">.*?</div>', re.DOTALL)
                
                if pattern.search(content):
                    new_content = pattern.sub(NEW_MENU, content)
                    if new_content != content:
                        with open(path, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                        print(f"Updated: {path}")

if __name__ == "__main__":
    target_dir = r"c:\Users\mackh\OneDrive\デスクトップ\files\Developer\MusicLiver05"
    update_html_files(target_dir)

File: update_menus.js
const fs = require('fs');
const path = require('path');

const NEW_MENU = `    <div class="hamburger-menu-content" id="hamburgerMenu">
      <a href="index.html">トップページ</a>
      <a href="schedule.html">全国のライブスケジュール一覧</a>
      <a href="live.html">現在演奏中のミュージシャン</a>
      <a href="artists.html">アーティスト一覧</a>
      <a href="admin_listener.html">マイページ</a>
    </div>`;

function updateHtmlFiles(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        
        if (stat.isDirectory()) {
            if (file !== 'node_modules' && file !== '.git' && file !== '.vercel') {
                updateHtmlFiles(fullPath);
            }
        } else if (file.endsWith('.html')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            const pattern = /<div class="hamburger-menu-content" id="hamburgerMenu">[\s\S]*?<\/div>/;
            
            if (pattern.test(content)) {
                const newContent = content.replace(pattern, NEW_MENU);
                if (newContent !== content) {
                    fs.writeFileSync(fullPath, newContent, 'utf8');
                    console.log(`Updated: ${fullPath}`);
                }
            }
        }
    }
}

updateHtmlFiles(__dirname);
